### ISSUE TITLE
Client.launch_cmd_cloudinit_option_reads_stdin_ok test fails in CI

### ISSUE THREAD

**Issue by Saviq** [canonical/multipass#651]
Now we got CI back, the below [happened there](https://travis-ci.org/CanonicalLtd/multipass/jobs/500005982):

```
[ RUN      ] Client.launch_cmd_cloudinit_option_reads_stdin_ok
/root/build_multipass/parts/multipass/src/tests/test_cli_client.cpp:233: Failure
Value of: send_command({"launch", "--cloud-init", "-"})
Expected: is equal to 0
  Actual: 1 (of type int)
[  FAILED  ] Client.launch_cmd_cloudinit_option_reads_stdin_ok (18 ms)
```

**Comment by gerboland** [comment]
Problem is that CI runs tests on a terminal with no tty connected, unlike running test on our own machines. Multipass has internal checks to change behaviour depending on if tty around or not.

I don't see a way to trick these multipass internal checks, without a bit of refactoring. Branch attached with what I think is least worst solution

**Comment by townsend2010** [comment]
Closed by #659.
